src/treebuilders/MapCalculator.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * MRCPP, a numerical library based on multiresolution analysis and
+ * the multiwavelet basis which provide low-scaling algorithms as well as
+ * rigorous error control in numerical computations.
+ * Copyright (C) 2020 Stig Rune Jensen, Jonas Juselius, Luca Frediani and contributors.
+ *
+ * This file is part of MRCPP.
+ *
+ * MRCPP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MRCPP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+#pragma once
+
+#include "TreeCalculator.h"
+
+namespace mrcpp {
+
+template <int D> class MapCalculator final : public TreeCalculator<D> {
+public:
+    MapCalculator(std::function<double(double)> fm, FunctionTree<D> &inp)
```

Comment:
Can you add a typedef or template alias for the function type? 
```c++
template <typename T, typename U>
using FMap_ = std::function<T(U)>;

typedef FMap_<double, double> FMap;
```

Reply:
And why would that be better? I'm in general not so keen on typedefs. I find they make the code more obscure to read.

Reply:
At any rate, where should the typedef be defined?

Reply:
If I want to change the input and output types of the function I change the `typedef` not N  different places. I don't know where you should put the definition.

Reply:
@stigrj any recommendations about where to ad the typedef?

Reply:
I guess that could be `api/constants.h` or `api/mrcpp_declarations.h`

Reply:
Done. I see there is a lot of `std::function<double(const  coord<D> r)>`
Should it undergo the same typdef-ification? (not in this PR)